src/Job.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * A job for a Daemon to complete.
+ */
+export class Job {
+	public incompletePrerequisites: Set<Job>
```

Comment:
By incomplete I mean unfinished. Incomplete prerequisites refers to all prerequisite jobs that are not yet complete (because `Job` forgets about each prerequisite as soon as its complete).
Would `unfinishedPrerequisites` or `incompletePrerequisiteJobs` be better?

Reply:
> `Job` forgets about each prerequisite as soon as its complete

Hmm this means that Job is a mutable object, which can lead to code that's hard to reason about. It would be better if a given Job is frozen once created, and whoever is using it (like the Orderer) is responsible for tracking what's complete.

Reply:
Why do you think that a `Job` being mutable would make the code harder to reason about?

Reply:
In general, when you have an object that can change, you always have to think about all the states it can be in. On the other hand, an immutable object needs to be constructed correctly, and then it's always correct. That's why a lot of functional programming languages require immutable data, and in JS we use `const` instead of `let` unless needed.

Of course we're not really trying to write a pure functional program here, so mutation is ok 😂. The distinction is between data, and machinery that acts on data. `Job` is data, so it should be unchangeable. Meanwhile JobOrderer has a bunch of private fields that are like the temporary storage of an algorithm, so it's totally fine to use that as scratch space.

Practically, you can fix this by having JobOrderer copy the data it needs and then modify the copy, and keep the modified stuff private. I havent't read the code in detail since the changes though, so let me know if you'd like more specific advice.